game-core/src/main/java/games/strategy/util/Version.java

```diff
@@ -0,0 +1,192 @@
+      return;
+    }
+    throw new IllegalArgumentException("Invalid version String: " + version);
+  }
+
+  /**
+   * Returns the exact and full version number. For example, if we specify: <code>
+   * new Version(1.2.3.4.5).getMicro == 4; // true
+   * new Version(1.2.3.4.5).toString().equals("1.2.3.4"); // true
+   * new Version(1.2.3.4.5).getExactVersion.equals("1.2.3.4.5"); // true
+   * </code>
+   */
+  public String getExactVersion() {
+    // in case of deserialization, exactVersion may be null, in which case toString() it.
+    return exactVersion != null ? exactVersion : toString();
+  }
+
+  /** Returns the major version number. */
+  public int getMajor() {
+    return m_major;
+  }
+
+  /** Returns the minor version number. */
+  public int getMinor() {
+    return m_minor;
+  }
+
+  /** Returns the point version number. */
+  public int getPoint() {
+    return m_point;
+  }
+
+  /** Returns the micro version number. */
+  public int getMicro() {
+    return m_micro;
+  }
+
+  @Override
+  public boolean equals(final @Nullable Object o) {
+    return o instanceof Version && compareTo((Version) o) == 0;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(m_major, m_minor, m_point, m_micro);
+  }
+
+  @Override
+  public int compareTo(final Version other) {
+    checkNotNull(other);
+
+    return Comparator.comparingInt(Version::getMajor)
+        .thenComparingInt(Version::getMinor)
+        .thenComparingInt(Version::getPoint)
+        .thenComparingInt(Version::getMicro)
+        .compare(this, other);
+  }
+
+  /**
+   * Indicates this version is greater than the specified version.
+   *
+   * @param other The version to compare.
+   * @return {@code true} if this version is greater than the specified version; otherwise {@code
+   *     false}.
+   */
+  public boolean isGreaterThan(final Version other) {
+    checkNotNull(other);
+
+    return compareTo(other) > 0;
+  }
+
+  /**
+   * Indicates this version is greater than or equal to the specified version.
+   *
+   * @param other The version to compare.
+   * @return {@code true} if this version is greater than or equal to the specified version;
+   *     otherwise {@code false}.
+   */
+  public boolean isGreaterThanOrEqualTo(final Version other) {
+    checkNotNull(other);
+
+    return compareTo(other) >= 0;
+  }
+
+  /**
+   * Indicates this version is less than the specified version.
+   *
+   * @param other The version to compare.
+   * @return {@code true} if this version is less than the specified version; otherwise {@code
+   *     false}.
+   */
+  public boolean isLessThan(final Version other) {
+    checkNotNull(other);
+
+    return compareTo(other) < 0;
+  }
+
+  /**
+   * Returns a new version with the major, minor, and point versions from this instance and the
+   * specified micro version.
+   */
+  public Version withMicro(final int micro) {
+    return new Version(m_major, m_minor, m_point, micro);
+  }
```

Comment:
This whole part as well as the constructors can be removed IMO, just to be sure no one actually tries to do something about this. 🤷‍♂ 

Just because it's deprecated doesn't mean people won't actually try to use it

Reply:
Not willing to modify this.
1. Would like diffs to be minimal
2. Waste of time reviewing changes here and equally responding to any potential feedback, and I would be lost for words if we introduced any further problems and then had to spend yet more time on *that*. We need to value our time highly, and keep that in mind when considering our review bar and how much stuff we need to get done. Failures is an option, the community is shrinking, there are new alternatives for players being made, we need to get stuff done.

> Just because it's deprecated doesn't mean people won't actually try to use it

By that reasoning, folks might try to append to it as well. Hopefully the big strike-through makes it clear this is deprecated.